odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -0,0 +1,1064 @@
+        the record name_get.
+        """
+        properties_values = (self.message_1 | self.message_3).read(['custom_properties'])
+
+        self.assertEqual(len(properties_values), 2)
+
+        properties_message_1 = properties_values[0]['custom_properties']
+        properties_message_3 = properties_values[1]['custom_properties']
+
+        self.assertTrue(isinstance(properties_message_1, list))
+        self.assertTrue(isinstance(properties_message_3, list))
+
+        self.assertEqual(len(properties_message_1), 2, msg="Message 1 has 2 properties")
+        self.assertEqual(len(properties_message_3), 1, msg="Message 3 has 1 property")
+
+        self.assertEqual(
+            properties_message_1[0]['id'], 'name',
+            msg='First message 1 property should be "name"')
+        self.assertEqual(
+            properties_message_1[1]['id'], 'partner_id',
+            msg='First message 1 property should be "partner_id"')
+        self.assertEqual(
+            properties_message_3[0]['id'], 'state',
+            msg='First message 3 property should be "state"')
+
+        many2one_property = properties_message_1[1]
+        self.assertEqual(
+            many2one_property['string'], 'Partner',
+            msg='Parent property definition must be present when reading child')
+        self.assertEqual(
+            many2one_property['type'], 'many2one',
+            msg='Parent property definition must be present when reading child')
+        self.assertEqual(
+            many2one_property['model'], 'res.partner',
+            msg='Parent property definition must be present when reading child')
+        self.assertEqual(many2one_property['value'], (self.partner.id, self.partner.display_name))
+
+        # disable the name_get
+        properties_values = (self.message_1 | self.message_3).read(['custom_properties'], load=None)
+        many2one_property = properties_values[0]['custom_properties'][1]
+
+        self.assertEqual(
+            many2one_property['value'], self.partner.id,
+            msg='If name_get is disable, should only return the record id')
+
+    def test_properties_field_many2one_basic(self):
+        """Test the basic (read, write...) of the many2one property."""
+        partner = self.env['res.partner'].create({'name': 'Test Partner'})
+        self.message_2.custom_properties = {'partner_id': [partner.id, 'res.partner']}
+
+        self.assertFalse(self.message_2.custom_properties[0]['value'])
+        self.assertEqual(self.message_2.custom_properties[1]['value'], partner.id)
+        sql_values = self._get_sql_properties(self.message_2)
+        self.assertEqual(sql_values, {'name': None, 'partner_id': [partner.id, 'res.partner']})
+
+        # read the many2one
+        properties = self.message_2.read(['custom_properties'])[0]['custom_properties']
+        self.assertEqual(properties[1]['value'], (partner.id, partner.display_name))
```

Comment:
I think I've mentioned this already, but isn't it strange that you don't get the same thing when you call read on the record or when you access the record field value?